src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -24,22 +24,118 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
+declare(strict_types=1);
+
 namespace PrestaShopBundle\Controller\Admin\Sell\Catalog;
 
 use Exception;
 use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\FeatureConstraintException;
 use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\FeatureNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Feature\Query\GetFeatureForEditing;
+use PrestaShopBundle\Bridge\AdminController\Action\HeaderToolbarAction;
+use PrestaShopBundle\Bridge\AdminController\Action\ListBulkAction;
+use PrestaShopBundle\Bridge\AdminController\Action\ListHeaderToolbarAction;
+use PrestaShopBundle\Bridge\AdminController\Action\ListRowAction;
+use PrestaShopBundle\Bridge\AdminController\AdminControllerTrait;
+use PrestaShopBundle\Bridge\AdminController\ControllerConfiguration;
+use PrestaShopBundle\Bridge\AdminController\Field\Field;
+use PrestaShopBundle\Bridge\AdminController\LegacyControllerBridgeInterface;
+use PrestaShopBundle\Bridge\AdminController\LegacyListControllerBridgeInterface;
+use PrestaShopBundle\Bridge\Helper\HelperListConfiguration;
+use PrestaShopBundle\Bridge\Helper\HelperListCustomizer\HelperListFeatureBridge;
+use PrestaShopBundle\Bridge\Smarty\SmartyTrait;
 use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
 use PrestaShopBundle\Security\Annotation\AdminSecurity;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Tools;
 
 /**
  * Controller responsible for "Sell > Catalog > Attributes & Features > Features" page
  */
-class FeatureController extends FrameworkBundleAdminController
+class FeatureController extends FrameworkBundleAdminController implements LegacyControllerBridgeInterface, LegacyListControllerBridgeInterface
 {
+    use AdminControllerTrait;
+    use SmartyTrait;
+
+    /**
+     * This parameter is needed by legacy hook, so we can't remove it.
+     *
+     * @var string
+     */
+    public $php_self;
```

Comment:
If we handle the bridge/proxy/decorator objected I suggested this could be removed from the symfony controller